src/components/App/App.js

```diff
@@ -0,0 +1,250 @@
+        if (res.token) {
+          localStorage.setItem("token", res.token);
+          setIsLoggedIn(true);
+          setLoginErr("");
+          setActiveModal("");
+          navigate("/tasks");
+        }
+      })
+      .catch((err) => {
+        console.log("Login response:", err);
+        setLoginErr(err.message);
+      });
+  };
+
+  const handleLogout = () => {
+    localStorage.removeItem("token");
+    setIsLoggedIn(false);
+    navigate("/");
+  };
+
+  const openLoginModal = () => {
+    setActiveModal("login");
+  };
+
+  const openRegisterModal = () => {
+    setActiveModal("register");
+  };
+
+  return (
+    <>
+      <Header
+        isLoggedIn={isLoggedIn}
+        onLogout={handleLogout}
+        openLoginModal={openLoginModal}
+        openRegisterModal={openRegisterModal}
+        handleLogout={handleLogout}
+      />
+
+      <Routes>
+        <Route path="/" element={<Home />} />
+        <Route
+          path="/register"
+          element={<Register onRegister={handleRegistration} />}
+        />
+        <Route path="/login" element={<Login onLogin={handleLogin} />} />
+        <Route
+          path="/tasks"
+          element={
+            <PrivateRoute isLoggedIn={isLoggedIn}>
+              <TaskList tasks={tasks} onTaskClick={handleSelectedTask} />
+            </PrivateRoute>
+          }
+        />
+        <Route
+          path="/create-task"
+          element={
+            <CreateTask
+              onCloseModal={handleCloseModal}
+              onAddTask={handleCreateTask}
+              isLoading={isLoading}
+            />
+          }
+        />
+        <Route
+          path="/update-task/:taskId"
+          element={
+            <UpdateTask
+              task={selectedTask}
+              onCloseModal={handleCloseModal}
+              onUpdateTask={handleUpdateTask}
+              isLoading={isLoading}
+            />
+          }
+        />
+        <Route
+          path="/delete-task/:taskId"
+          element={
+            <DeleteTask
+              task={selectedTask}
+              onCloseModal={handleCloseModal}
+              onDeleteTask={handleDeleteTask}
+              isLoading={isLoading}
+            />
+          }
+        />
+      </Routes>
+
+      {activeModal === "preview" && (
+        <Task task={selectedTask} onCloseModal={handleCloseModal} />
+      )}
+      {activeModal === "login" && (
+        <LoginModal
+          setActiveModal={setActiveModal}
+          loginErr={loginErr}
+          handleLogin={handleLogin}
+        />
+      )}
+      {activeModal === "register" && (
+        <RegisterModal
+          setActiveModal={setActiveModal}
+          registerErr={registerErr}
+          handleRegistration={handleRegistration}
+        />
+      )}
+      <Footer />
+    </>
+  );
+};
+
+const App = () => {
+  const [tasks, setTasks] = useState([]);
+  const [isLoggedIn, setIsLoggedIn] = useState(
+    localStorage.getItem("token") ? true : false
```

Comment:
`token` in `localStorage` doesn't mean that you can log the user in. The token can be expired/invalid. You need to check if it's still **fresh** and valid on the server

Reply:
Thanks for your feedback. I believe that I have already addressed the issue you brought up. A. I’ve added a checkRequests Higher Order Component (HOC) on the frontend. This HOC intercepts every fetch request, checks the response status, and if it’s 401 (Unauthorized), it removes the token from the localStorage and redirects the user to the login page. B. I’ve also modified the global fetch function to handle HTTP request errors. If a fetch request fails, the function checks the HTTP response status. If it’s 401, the function removes the ‘token’ from the localStorage and redirects the user to the login page. C. On the backend, I’ve included a middleware that verifies the token sent with each request. If the token is not present or is invalid, it responds with a 401 status, which triggers the interceptor on the frontend to redirect the user to the login page. These changes ensure the token is still fresh and valid on the server before allowing a user to log in, addressing your concern about checking the validity of the token on the server side. 
 
![A](https://github.com/iankamar/task-manager-frontend/assets/95672055/a1ce04eb-827c-4676-94d2-d367b77903a3)
![B](https://github.com/iankamar/task-manager-frontend/assets/95672055/eae03df2-f589-48be-998f-ad0c98613b57)
![C](https://github.com/iankamar/task-manager-frontend/assets/95672055/ed41a84e-2659-4918-95b7-693196d33974)